### PR TITLE
Collaborative edition example

### PR DESCRIPTION
**Pull request by cgestes**
This is an example of a collaborative editor for Fleather.

Should we extract the logic somehow to make it reusable?

Note that there is a bug that make the focus switch between the two editors in the example.

Feedback welcome :)

### CODE REVIEW

File: packages/fleather/example/lib/src/home.dart [hunk 1]
```diff
@@ -95,7 +96,7 @@ class _HomePageState extends State<HomePage> {
           color: Colors.grey.shade800,
           child: _buildMenuBar(context),
         ),
-        body: _buildWelcomeEditor(context),
+        body: CollaborativeEditionDemo(), //_buildWelcomeEditor(context),
```
Comment (cgestes) [inline packages/fleather/example/lib/src/home.dart:99]: needs to be reverted once finished.

**Comment by Amir-P** [comment]
Thanks for contributing @cgestes . It's good to see you here!
I'm a bit busy these days but I'll try to check it in the next week.

**Comment by cgestes** [comment]
No worries. take your time.

There are a few commits that if they make sense can go in separated PR.

most notably the last one about stealing focus, I tried on Linux and Android and it doesnt seem to have any bad impact.

**Comment by cgestes** [comment]
replaced the demo in this PR by a repository: https://github.com/jotshq/fleather_demo_collab

**Pull request closed without merging**